radis/rag/factories.py

```diff
@@ -0,0 +1,62 @@
+from typing import Generic, TypeVar
+
+import factory
+from faker import Faker
+
+from .models import Answer, Question, RagJob, RagReportInstance, RagTask
+
+T = TypeVar("T")
+
+fake = Faker()
+
+
+class BaseDjangoModelFactory(Generic[T], factory.django.DjangoModelFactory):
+    @classmethod
+    def create(cls, *args, **kwargs) -> T:
+        return super().create(*args, **kwargs)
+
+
+SearchProviders = ("OpenSearch", "Vespa", "Elasticsearch")
+PatientSexes = ["", "M", "F"]
+
+
+class RagJobFactory(BaseDjangoModelFactory):
+    class Meta:
+        model = RagJob
+
+    title = factory.Faker("sentence", nb_words=3)
+    provider = factory.Faker("random_element", elements=SearchProviders)
+    group = factory.SubFactory("adit_radis_shared.accounts.factories.GroupFactory")
+    query = factory.Faker("word")
+    language = factory.SubFactory("radis.reports.factories.LanguageFactory")
+    # TODO: handle modalities
```

Comment:
https://www.perplexity.ai/search/with-factoryboy-in-django-i-wa-N72YYWp0RZuxSYqm5E1LuA

Perplexity.ai ist übrigens deutlich besser als ChatGPT.